api/src/main/java/com/basistech/rosette/api/RosetteAPI.java

```diff
@@ -339,16 +336,33 @@ public PingResponse ping() throws IOException, RosetteAPIException {
     /**
      * Matches 2 names and returns a score in NameMatchingResponse.
      *
-     * @param request request object with the names.
+     * @param request request
      * @return response
      * @throws RosetteAPIException Rosette specific exception
      * @throws IOException         General IO exception
+     *
+     * @deprecated replaced by {@link #getNameSimilarity(Name, Name) getNameSimilarity}
      */
+    @Deprecated
     public NameSimilarityResponse getNameSimilarity(NameSimilarityRequest request) throws RosetteAPIException, IOException {
         return sendPostRequest(request, urlBase + NAME_SIMILARITY_SERVICE_PATH, NameSimilarityResponse.class);
```

Comment:
Should just call the new method? `getNameSimilarity(request.getName1(), request.getName2())`